### PR TITLE
Expand filename shown in backtrace

### PR DESCRIPTION
**Pull request by rocky**
Towards addressing problem in screenshot 1. Regression tests fail, so more work is needed.

### CODE REVIEW

File: command/backtrace.sh [hunk 1]
```diff
@@ -124,12 +124,17 @@ function _Dbg_do_backtrace {
...
     if (( frame_start == 0 )) ; then
 	((count--)) ;
 	adjusted_pos=$(_Dbg_frame_adjusted_pos 0)
-	filename=$(_Dbg_file_canonic "${BASH_SOURCE[$adjusted_pos]}")
+	filename="${BASH_SOURCE[$adjusted_pos]}"
+	resolved_filename="${_Dbg_file2canonic[$filename]}"
+	if [[ ! -z "${resolved_filename}" ]] ; then
+	    filename="$resolved_filename"
+	else
+	    filename=$(_Dbg_file_canonic "$filename")
```
Comment (jansorg) [inline command/backtrace.sh]: Probably needs quotes for whitespace handling.
```suggestion
	    filename="$(_Dbg_file_canonic "$filename")"
```

File: command/backtrace.sh [hunk 2]
```diff
@@ -124,12 +124,17 @@ function _Dbg_do_backtrace {
...
     typeset    filename
     typeset -i adjusted_pos
-    # Position 0 is special in that get the line number not from the
-    # stack but ultimately from LINENO which was saved in the hook call.
     if (( frame_start == 0 )) ; then
 	((count--)) ;
 	adjusted_pos=$(_Dbg_frame_adjusted_pos 0)
-	filename=$(_Dbg_file_canonic "${BASH_SOURCE[$adjusted_pos]}")
+	filename="${BASH_SOURCE[$adjusted_pos]}"
+	resolved_filename="${_Dbg_file2canonic[$filename]}"
```
Comment (jansorg) [inline command/backtrace.sh:131]: should this have a `local` or `typeset`?

File: lib/file.sh [hunk 3]
```diff
@@ -98,21 +98,26 @@ function _Dbg_resolve_expand_filename {
...
   else
     # Resolve file using _Dbg_dir
     typeset -i n=${#_Dbg_dir[@]}
     typeset -i i
     for (( i=0 ; i < n; i++ )) ; do
-      typeset basename="${_Dbg_dir[i]}"
+      typeset dirname="${_Dbg_dir[i]}"
       if [[  "$basename" == '\$cdir' ]] ; then
-	basename=$_Dbg_cdir
+	dirname=$_Dbg_cdir
```
Comment (jansorg) [inline lib/file.sh]: quotes?

File: lib/file.sh [hunk 4]
```diff
@@ -98,21 +98,26 @@ function _Dbg_resolve_expand_filename {
...
       elif [[ "$basename" == '\$cwd' ]] ; then
-	basename=$(pwd)
+	dirname=$(pwd)
```
Comment (jansorg) [inline lib/file.sh]: quotes?

**Review by jansorg — COMMENTED** [review]
Thank you for working on this!
When I run this branch with my test sources, then I'm seeing this error.
I looked at it, but couldn't spot the problem.
For my review, only minor notes so far.

```
[jansorg@Island sources]$ ../../bashdb --style colorful -L ../.. main.bash 
bash debugger, bashdb, release 5.2-1.1.2

Copyright 2002-2004, 2006-2012, 2014, 2016-2019, 2021, 2023-2024 Rocky Bernstein
This is free software, covered by the GNU General Public License, and you are
welcome to change it and/or distribute copies of it under certain conditions.

** Internal debug error _Dbg_is_file(): file argument null
../../lib/filecache.sh: Zeile 247: _Dbg_filenames["$fullname"]: Falscher Feldindex.
hello world
Debugged program terminated normally. Use q to quit or R to restart.
bashdb<0> 

```

File: command/backtrace.sh (hunk 2 above)
Comment (rocky) [reply command/backtrace.sh:131]: Yes, you are correct. Should be corrected now.

File: lib/file.sh (hunk 3 above)
Comment (rocky) [reply lib/file.sh]: Yes, you are correct. Should be handled now.

File: lib/file.sh (hunk 4 above)
Comment (rocky) [reply lib/file.sh]: Yes, you are correct; this should be handled now.

**Comment by rocky** [comment]
Superseded by #54

**Pull request closed without merging**